### ISSUE TITLE
LinearMouse not working in macos monterey

### ISSUE THREAD

**Issue by yahlor** [linearmouse/linearmouse#17]
linear acceleration, reverse scrolling, linear movement doesn't seem to work in macos monterey, however, the side buttons do

**Comment by lujjjh** [comment]
Could you try the following steps?

1. Open System Preferences.
2. Click Security & Privacy.
3. Click Accessibility.
4. Delete and re-add LinearMouse and make sure LinearMouse is selected.

**Comment by yahlor** [comment]
nope still doesn't work lol

**Comment by lujjjh** [comment]
Cannot reproduce on macOS Monterey 12.0.1 (21A559).

Have you previously used LinearMouse with the same mouse device on macOS Big Sur? Does it work?

**Comment by yahlor** [comment]
No, I've only started using LinearMouse in macOS Monterey with my Logitech g203

**Comment by RMoDOOM** [comment]
@yahlor Do you have Logitech Options installed? I was having the same issue with my MX Master 3 and I was curious if Options was causing it. I uninstalled and LinearMouse immediately started working.

**Comment by yahlor** [comment]
I use the G Hub software, does it also affect the trackpad as the trackpad also does not have linear acceleration

**Comment by lujjjh** [comment]
> [...] does it also affect the trackpad [...]

@yahlor No, it only affects mouses.

LinearMouse would reguard a device as a mouse if its scroll wheel is not "continuous":

https://github.com/lujjjh/LinearMouse/blob/ea4df3ac0e471da5b08aac011659267c05eebd7c/LinearMouse/MouseWheelEvent.swift#L15-L18

I'm thinking about two possible reasons:

1. Logitech Options or G Hub intercepted mouse wheel events which are not passed through to LinearMouse. (However, I couldn't repro it with Logitech Options and ERGO M575.)
2. Logitech Options or G Hub modifies the scroll wheel to "continuous" and LinearMouse regards the mouse as a trackpad. (I'll add some logs in LinearMouse.)

Could you please check if [Scroll Reverser](https://pilotmoon.com/scrollreverser/) or some other mouse utilities work?

**Comment by yahlor** [comment]
Scroll reverser works

**Comment by lujjjh** [comment]
@RMoDOOM I just found something in Logitech Options:

![image](https://user-images.githubusercontent.com/3000535/143237428-e3505409-e9f3-4832-a56d-7ae6f09b384c.png)

Have you tried disabling smooth scrolling in Logitech Options? With my mouse, there is no such an option so I couldn't test it.

@yahlor I'm not sure if there are such similar options in G Hub. (I couldn't even open G Hub after installation.)

Also, I've checked the source code from Scroll Reverser. It uses a more [complicated algorithm](https://github.com/pilotmoon/Scroll-Reverser/blob/f81281ec262daad25dd6244f334ef7188070ce71/MouseTap.m#L124-L144) to detect if an event comes from a mouse or trackpad and the algorithm seems not to work very well in Monterey. So I would wait and see if there is more elegant approach to fix this issue.

**Comment by yahlor** [comment]
G Hub does not. have this option

**Comment by tdwjl** [comment]
Having same issue on MX ergo, logi options doesn't have smooth scroll checkbox. https://smooze.co/ seems to work, but would like to stick with linearmouse if possible.

**Comment by tdwjl** [comment]
update: uninstalled logi options, uninstalled, restarted, reinstalled and it works! ty for this app.

**Comment by lujjjh** [comment]
In #82, I added a new experimental way to detect mouse events.

If this issue still persists, please try [this build](https://github.com/linearmouse/linearmouse/actions/runs/1833282969) and turn on ‘Enable experimental mouse detector’.

<img width="350" alt="image" src="https://user-images.githubusercontent.com/3000535/153704868-76fc99cc-ac16-45dc-bb35-97eb95b6ceca.png">

Thank you!

**Comment by yahlor** [comment]
Thanks

**Comment by lujjjh** [comment]
Starting from v0.6.0-beta.5, LinearMouse adjusted the place to insert the EventTap. I believe this issue has been resolved.

Please feel free to reopen it if you have any further questions.
